query-engine/prisma-models/src/builders/internal_dm_builder.rs

```diff
@@ -220,17 +222,16 @@ fn index_builders(model: &dml::Model) -> Vec<IndexBuilder> {
     model
         .indices
         .iter()
-        .filter(|i| i.fields.len() > 1 && model.is_compound_index_supported(i)) // @@unique for 1 field are transformed to is_unique instead
-        .filter(|i| i.fields.iter().all(|f| f.path.len() <= 1)) // TODO: we do not take indices with composite fields for now
-        .map(|i| IndexBuilder {
-            name: i.name.clone(),
-            fields: i
+        .filter(|i| model.is_compound_index_supported(i))
+        .map(|idx| IndexBuilder {
+            name: idx.name.clone(),
+            field_paths: idx
```

Comment:
This looks like a method we could add onto the `IndexDefinition`

Reply:
Same case as some previous comment, you will fraction and create a method that is used _only once_ in a given and specific path and create a single function that will replace it, I see no benefit for it and it will complicate navigation. I will say this again, a problem I am perceiving in the code is the indiscriminated use of small functions used only once. That is not good and creates noise.